Locale.parse('zh_CN') raises a TypeError on Babel 2.0 with Python 3.3 and 3.4.

```
Python 3.4.3 (default, Apr 13 2015, 22:30:47)
[GCC 4.6.3] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import babel
>>> babel.__version__
'2.0'
>>> from babel.core import Locale
>>> Locale.parse('zh_CN')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/venv/py34/lib/python3.4/site-packages/babel/core.py", line 286, in parse
    language = get_global('language_aliases').get(language, language)
  File "/venv/py34/lib/python3.4/site-packages/babel/core.py", line 58, in get_global
    _global_data = pickle.load(fileobj)
TypeError: an integer is required (got type str)
>>>
```
`Locale.parse('zh_CN')` should work and load `zh_CN` localedata.

On Python 2.7, it works fine.
```
Python 2.7.3 (default, Jun 22 2015, 19:33:41)
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import babel
>>> babel.__version__
'2.0'
>>> from babel.core import Locale
>>> Locale.parse('zh_CN')
Locale('zh', territory='CN', script='Hans')
>>>
```

Ran into this one as well on Windows & 3.4: https://ci.appveyor.com/project/ericholscher/sphinx-autoapi/build/1.0.5

```
=================================== ERRORS ==================================== 
_________________ ERROR collecting tests/test_integration.py __________________ 
tests\test_integration.py:9: in <module> 
    from sphinx.application import Sphinx 
.tox\py34\lib\site-packages\sphinx\application.py:37: in <module> 
    from sphinx.builders import BUILTIN_BUILDERS 
.tox\py34\lib\site-packages\sphinx\builders\__init__.py:23: in <module> 
    from sphinx.util import i18n, path_stabilize 
.tox\py34\lib\site-packages\sphinx\util\i18n.py:15: in <module> 
    from babel.messages.pofile import read_po 
.tox\py34\lib\site-packages\babel\messages\__init__.py:12: in <module> 
    from babel.messages.catalog import * 
.tox\py34\lib\site-packages\babel\messages\catalog.py:23: in <module> 
    from babel.dates import format_datetime 
.tox\py34\lib\site-packages\babel\dates.py:28: in <module>
    from babel.util import UTC, LOCALTZ
.tox\py34\lib\site-packages\babel\util.py:278: in <module>
    from babel import localtime 
.tox\py34\lib\site-packages\babel\localtime\__init__.py:21: in <module> 
    from babel.localtime._win32 import _get_localzone 
.tox\py34\lib\site-packages\babel\localtime\_win32.py:18: in <module> 
    tz_names = get_global('windows_zone_mapping') 
.tox\py34\lib\site-packages\babel\core.py:58: in get_global 
    _global_data = pickle.load(fileobj) 
E   TypeError: an integer is required (got type str) 
```

--- a/CHANGES
+++ b/CHANGES
@@ -1,6 +1,11 @@
 Babel Changelog
 ===============
 
+Version 2.1
+-----------
+(bugfix release, to be defined)
+- Fix Locale.parse using ``global.dat`` incompatible types (https://github.com/mitsuhiko/babel/issues/174)
+
 Version 2.0
 -----------
 

--- a/babel/__init__.py
+++ b/babel/__init__.py
@@ -21,4 +21,4 @@ from babel.core import UnknownLocaleError, Locale, default_locale, \
      negotiate_locale, parse_locale, get_locale_identifier
 
 
-__version__ = '2.0'
+__version__ = '2.1'

--- a/babel/numbers.py
+++ b/babel/numbers.py
@@ -134,6 +134,10 @@ def get_territory_currencies(territory, start_date=None, end_date=None,
 
     result = []
     for currency_code, start, end, is_tender in curs:
+        if start:
+            start = date_(*start)
+        if end:
+            end = date_(*end)
         if ((is_tender and tender) or \
             (not is_tender and non_tender)) and _is_active(start, end):
             if include_details:

--- a/scripts/import_cldr.py
+++ b/scripts/import_cldr.py
@@ -21,8 +21,6 @@ try:
 except ImportError:
     from xml.etree import ElementTree
 
-from datetime import date
-
 # Make sure we're using Babel source, and not some previously installed version
 sys.path.insert(0, os.path.join(os.path.dirname(sys.argv[0]), '..'))
 
@@ -103,12 +101,12 @@ def _parse_currency_date(s):
     if not s:
         return None
     parts = s.split('-', 2)
-    return date(*map(int, parts + [1] * (3 - len(parts))))
+    return tuple(map(int, parts + [1] * (3 - len(parts))))
 
 
 def _currency_sort_key(tup):
     code, start, end, tender = tup
-    return int(not tender), start or date(1, 1, 1)
+    return int(not tender), start or (1, 1, 1)
 
 
 def main():

--- a/setup.py
+++ b/setup.py
@@ -32,7 +32,7 @@ class import_cldr(Command):
 
 setup(
     name='Babel',
-    version='2.0',
+    version='2.1',
     description='Internationalization utilities',
     long_description=\
 """A collection of tools for internationalizing Python applications.""",

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -269,3 +269,29 @@ def test_parse_locale():
     assert core.parse_locale('en_US.UTF-8') == ('en', 'US', None, None)
     assert (core.parse_locale('de_DE.iso885915@euro') ==
             ('de', 'DE', None, None))
+
+def test_compatible_classes_in_global_and_localedata():
+    # Use pickle module rather than cPickle since cPickle.Unpickler is a method
+    # on Python 2
+    import pickle
+
+    class Unpickler(pickle.Unpickler):
+        def find_class(self, module, name):
+            # *.dat files must have compatible classes between Python 2 and 3
+            if module.split('.')[0] == 'babel':
+                return pickle.Unpickler.find_class(self, module, name)
+            raise pickle.UnpicklingError("global '%s.%s' is forbidden" %
+                                         (module, name))
+
+    def load(filename):
+        with open(filename, 'rb') as f:
+            return Unpickler(f).load()
+
+    load('babel/global.dat')
+    load('babel/localedata/root.dat')
+    load('babel/localedata/en.dat')
+    load('babel/localedata/en_US.dat')
+    load('babel/localedata/en_US_POSIX.dat')
+    load('babel/localedata/zh_Hans_CN.dat')
+    load('babel/localedata/zh_Hant_TW.dat')
+    load('babel/localedata/es_419.dat')

